states/huntsman/ready.py

```diff
@@ -9,4 +9,20 @@ def on_enter(event_data):
 
     pocs.observatory.mount.unpark()
 
-    pocs.next_state = 'calibrating'
+    # Check if we need to foucs
+    if pocs.is_dark(horizon='focus') and pocs.observatory.require_coarse_focus():
+        pocs.next_state = 'coarse_focusing'
+
+    # Check if we should go straight to observing
+    elif pocs.is_dark(horizon='observe'):
+        pocs.next_state = 'scheduling'
+
+    # Don't need to focus, not dark enough to observe
+    elif pocs.observatory.past_midnight():
+        if pocs.is_dark(horizon='flat'):
+            pocs.next_state = 'twilight_flat_fielding'
+        else:
+            # We don't want to be waiting all day to take evening flats
+            pocs.next_state = 'parking'
+    else:
+        pocs.next_state = 'twilight_flat_fielding'
```

Comment:
This change fixes issue noted in comment above. Assumes `next_state` has already been set to `parking`.

```suggestion
    # Don't need to focus, not dark enough to observe
    else:
        if pocs.observatory.past_midnight():
            if pocs.is_dark(horizon='flat'):
                # After midnight and still dark enough to take flats
                pocs.next_state = 'twilight_flat_fielding'
        else:
            if not pocs.is_dark(horizon='focus'):
                # Before midnight and still light enough to take flats
                pocs.next_state = 'twilight_flat_fielding'
```